src/hooks/screen/useCreateChat.ts

```diff
@@ -42,17 +42,24 @@ const useCreateChat = () => {
       channel_type: 'messaging',
       channel: channelWithMember,
       created_at: response?.channel,
-      targetName: selectedUser?.user?.username || selectedUser?.user?.name,
-      targetImage: selectedUser?.user?.profilePicture || selectedUser?.user?.image
+      targetName: selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username,
+      targetImage:
+        selectedUser?.user?.profilePicture ||
+        selectedUser?.user?.image ||
+        selectedUser.profilePicture
     };
     const chatData = {
       channel: response?.channel,
       members: response?.members,
       appAdditionalData: {
         rawJson: targetRawJson,
         message: '',
-        targetName: selectedUser?.user?.username || selectedUser?.user?.name,
-        targetImage: selectedUser?.user?.profilePicture || selectedUser?.user?.image
+        targetName:
+          selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username,
+        targetImage:
+          selectedUser?.user?.profilePicture ||
+          selectedUser?.user?.image ||
+          selectedUser.profilePicture
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This is a repetition of the logic used in lines 45-49. To adhere to the DRY (Don't Repeat Yourself) principle, consider extracting this logic into a function that can be called in both places. Here's an example:

```diff
+ const getTargetName = (selectedUser) => selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username;
+ const getTargetImage = (selectedUser) => selectedUser?.user?.profilePicture || selectedUser?.user?.image || selectedUser.profilePicture;

- targetName: selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username,
+ targetName: getTargetName(selectedUser),
- targetImage: selectedUser?.user?.profilePicture || selectedUser?.user?.image || selectedUser.profilePicture
+ targetImage: getTargetImage(selectedUser)
```

And then use these functions again on lines 57-62. This will make your code more maintainable and easier to read.


<!-- This is an auto-generated comment by OSS CodeRabbit -->